modules/carto/src/api/layer-map.ts

```diff
@@ -135,7 +146,7 @@ function mergePropMaps(a: Record<string, any> = {}, b: Record<string, any> = {})
 }
 
 export function getLayer(
-  type: string,
+  type: LayerType,
```

Comment:
This is actually incorrect, `type` is used below for non-tile layers

Reply:
Perhaps we should rename `LayerType` to `TileLayerType` and have `type LayerType = 'point' | 'hexagon' ...`

Reply:
> Perhaps we should rename LayerType to TileLayerType and have type LayerType = 'point' | 'hexagon' ...

That sounds like a good plan!

What about:

```
type TileLayerType = 'raster' | 'mvt' | 'tileset' | 'quadbin' | 'h3' | 'heatmapTile';
type LayerType = 'point' | 'geojson' | 'grid' | 'heatmap' | 'hexagon' | 'hexagonId' | TileLayerType;
```

Reply:
Looks good, just `DocumentModeType` rather than `TableLayerType`